src/components/WeatherWidget.vue

```diff
@@ -45,6 +97,32 @@ const handleDayChange = (newDay: string) => {
     selectedDay.value = newDay;
   }
 };
+
+onMounted(async () => {
+  try {
+    isLoadingGeolocation.value = true;
+
+    coords.value = await getLocation();
+  } catch (err) {
+    geoLocationError.value = err as string;
+  } finally {
+    isLoadingGeolocation.value = false;
+  }
+
+  if (geoLocationError.value) return;
+
+  await fetchLocationKey("GET", {
+    params: { q: `${coords.value?.latitude},${coords.value?.longitude}` },
+  });
+
+  const locationKeyEndpoint = import.meta.env.VITE_MOCK
+    ? ""
+    : `/${locationKeyData.value?.key}`;
+
+  await fetchWeatherData("GET", {
+    url: `${API_URL}${endpoints.forecast}${locationKeyEndpoint}`,
```

Comment:
I agree, that looks weird but I was forced to do so because API needs to know the `locationKey` that I can retrieve from some other endpoint and than it is not passed as a param but it created completely different endpoint.

Maybe what I can do is to specify `baseUrl` instead of `url` in `useFetch` and than only extend it? What do you think about that?

Reply:
Yes, let's go for it. It would be cleaner.

Actually it's one of the major downsides of hooks like useFetch. You're stuck with the config that you previously specified and it's not as scalable. What I would do in my own application is to either call Axios methods directly (baseUrl would be specified on Axios initialization) or rewrite useFetch so that I doesn't require initial declaration and can be used immediately when you want to fetch some API data.

However in this case let's go with your suggestion since it will save us some time.

Reply:
Got it, thanks for the advice